### PR TITLE
[corlib] Adds Span tests

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: mcs/class/corlib/corlib_xtest.dll.sources [hunk 1]
```diff
@@ -145,3 +145,4 @@
 ../../../external/corefx/src/System.Text.Encoding/tests/EncodingTestHelpers.cs
 ../../../external/corefx/src/System.Text.Encoding/tests/NegativeEncodingTests.cs
 
+../../../external/corefx/src/System.Memory/tests/*.cs
```
Comment (akoeplinger) [inline mcs/class/corlib/corlib_xtest.dll.sources]: this won't include the subdirectories so it'll only include AllocationHelper.cs and TInt.cs

**Comment by marek-safar** [comment]
commit apidiff

**Comment by marek-safar** [comment]
commit apidiff

File: scripts/ci/run-test-default.sh [hunk 2]
```diff
@@ -85,6 +85,7 @@ ${TESTCMD} --label=System.Xaml --timeout=5m make -w -C mcs/class/System.Xaml run
 ${TESTCMD} --label=System.Net.Http --timeout=5m make -w -C mcs/class/System.Net.Http run-test
 ${TESTCMD} --label=System.Json --timeout=5m make -w -C mcs/class/System.Json run-test
 ${TESTCMD} --label=System.Threading.Tasks.Dataflow --timeout=5m make -w -C mcs/class/System.Threading.Tasks.Dataflow run-test
+if [[ ${label} == linux* ]]; then ${TESTCMD} --label=System.Runtime.CompilerServices.Unsafe-xunit --timeout=5m make -w -C mcs/class/System.Runtime.CompilerServices.Unsafe run-xunit-test; fi
```
Comment (akoeplinger) [inline scripts/ci/run-test-default.sh]: why are we only running this on Linux?

Comment (akoeplinger) [reply scripts/ci/run-test-default.sh]: actually, this won't run at all since don't have `label` with value `linux` (we have `debian-xx` and `ubuntu-xx`)

Comment (marek-safar) [reply scripts/ci/run-test-default.sh]: What should I use to skip on `Linux i386` only ?

Comment (akoeplinger) [reply scripts/ci/run-test-default.sh]: `ubuntu-1404-i386`. I assume this is due to some crash/bug on i386 we have tracked already?

Comment (marek-safar) [reply scripts/ci/run-test-default.sh]: yep, https://bugzilla.xamarin.com/show_bug.cgi?id=60865

Comment (akoeplinger) [reply scripts/ci/run-test-default.sh]: cool, can you pleas link that in the file?

**Comment by akoeplinger** [comment]
@monojenkins commit apidiff

**Pull request merged**